src/components/Workspace.jsx

```diff
@@ -314,21 +318,32 @@ export default class Workspace extends React.Component {
     );
   }
 
+  _handleSave() {
+    this.props.onSave();
+    return false;
+  }
+
   render() {
+    const handlers = {
+      SAVE: this._handleSave,
+    };
     return (
-      <div className="layout">
-        <AssignmentCreator />
-        <TopBar />
-        <NotificationList />
-        <div className="layout__columns">
-          <Instructions />
-          {this._renderInstructionsBar()}
-          <div className="workspace layout__main">
-            {this._renderEnvironment()}
+      <HotKeys handlers={handlers} keyMap={keyMap}>
```

Comment:
I don’t have a super-strong opinion about this, but I think it would be a little nicer to extract a small component to handle this along the lines of `<KeyboardHandler>` so we can avoid having too many implementation details in `<Workspace>`. Best to think of `<Workspace>`’s job as stitching together the top-level components of the application and leave all the details out of it.